sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -341,6 +351,10 @@ class AstBuilder extends DataTypeAstBuilder
         if Option(c.beginLabel()).isDefined &&
           c.beginLabel().multipartIdentifier().getText.toLowerCase(Locale.ROOT).equals(label)
         => true
+      case c: ForStatementContext
```

Comment:
To avoid code duplication, can we rewrite this case and the similar above ones as:

case c: WhileStatementContext
           | RepeatStatementContext
           | LoopStatementContext
           | ForStatementContext
        if Option(c.beginLabel()).isDefined &&
          c.beginLabel().multipartIdentifier().getText.toLowerCase(Locale.ROOT).equals(label)
        => true

Reply:
I agree we need this refactor, however is it's not trivial as these classes don't have a common superclass with `beginLabel() `method. 
The syntax for what you're suggesting would be
```
case c @ (WhileStatementContext
| RepeatStatementContext
| LoopStatementContext
| ForStatementContext)
if Option(c.beginLabel()).isDefined &&
c.beginLabel().multipartIdentifier().getText.toLowerCase(Locale.ROOT).equals(label)
=> true
```
however this doesn't compile, due to the above reason. To add the superclass with begin/end labels requires grammar changes, i think this is best done as a separate task.

Reply:
Sure, makes sense, probably best to consider adding a base class with common methods as part of a separate PR (not a priority), to avoid further code duplication.